### PR TITLE
Add metrics for certain operations in instrumentedHDFSFilesystem

### PR DESCRIPTION
**Pull request by yukuai518**

### CODE REVIEW

File: gobblin-core/src/main/java/gobblin/filesystem/InstrumentedHDFSFileSystem.java [hunk 1]
```diff
@@ -75,4 +147,143 @@ public void close()
     super.close();
     // Should print out statistics here
   }
+
+  /**
+   * Add timer metrics to {@link DistributedFileSystem#mkdir(Path, FsPermission)}
+   */
+  public boolean mkdir(Path f, FsPermission permission) throws IOException {
+    try (Closeable context = new TimerContextWithLog(mkdirTimer.time(), "mkdir", f, permission)) {
```
Comment (ibuenros) [inline gobblin-core/src/main/java/gobblin/filesystem/InstrumentedHDFSFileSystem.java]: nit: can simplify this as
```java
try (Closeable context = new TimerContextWithLog(mkdirTimer.time(), "mkdir", f, permission)) {
   return super.mkdir(f, permission);
}
```

Same for all the other methods.

File: gobblin-core/src/main/java/gobblin/filesystem/InstrumentedHDFSFileSystem.java [hunk 2]
```diff
@@ -75,4 +147,143 @@ public void close()
...
+    } catch (IOException e) {
+      throw e;
+    }
+  }
+
+  /**
+   * Indirect call to {@link InstrumentedHDFSFileSystem#mkdirs(Path, FsPermission)}
+   */
+  public boolean mkdirs(Path f) throws IOException {
+    return super.mkdirs(f);
```
Comment (ibuenros) [inline gobblin-core/src/main/java/gobblin/filesystem/InstrumentedHDFSFileSystem.java]: Please remove methods that just call `super`.

File: gobblin-core/src/main/java/gobblin/filesystem/InstrumentedHDFSFileSystem.java [hunk 3]
```diff
@@ -40,8 +53,67 @@
...
-
+  public static final String HDFS_METRIC_CONTEXT_NAME = "hdfsMetricContext";
   private static final String HDFS_SCHEME = "hdfs";
+  private MetricContext metricContext;
+
+  protected final Closer closer;
+
+  // Below are HDFS metrics
+  @VisibleForTesting
+  protected final Timer listStatusPathTimer;
```
Comment (ibuenros) [inline gobblin-core/src/main/java/gobblin/filesystem/InstrumentedHDFSFileSystem.java]: I think all of the `listStatus` type calls should use the same timer.

Comment (yukuai518) [inline gobblin-core/src/main/java/gobblin/filesystem/InstrumentedHDFSFileSystem.java]: There are a few listStatus may duplicate the timer count. 
listStatus(path[]) -> listStatus(path[], filter) ---> listStatus(path)
listStatus(path, filter) ________________________/

If we only care about the final operation listStatus(path), then only listStatusPathTimer should be inspected.

**Pull request merged**